internal/component/database_observability/mysql/collector/time.go

```diff
@@ -0,0 +1,62 @@
+package collector
+
+import "math"
+
+var picosecondsOverflowInSeconds = PicosecondsToSeconds(float64(math.MaxUint64))
+
+const (
+	picosecondsPerSecond      float64 = 1e12
+	millisecondsPerSecond     float64 = 1e3
+	millisecondsPerPicosecond float64 = 1e9
+	nanosecondsPerMillisecond float64 = 1e6
+)
+
+// CalculateWallTime calculates the wall-clock timestamp for an event.
+// The timerPicoseconds indicates event timing since server startup.
+// Since this value can overflow after approximately ~213 days (column type of bigint unsigned),
+// this function accounts for overflows by calculating the number of previous overflows and
+// compensating accordingly. Returns the timestamp in milliseconds when the event occurred.
```

Comment:
This is the same code as before, I've just extracted a function comment and appended time-related suffix to parameters.